sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -345,8 +343,6 @@ object ColumnPruning extends Rule[LogicalPlan] {
     // Prunes the unused columns from child of Aggregate/Window/Expand/Generate
     case a @ Aggregate(_, _, child) if (child.outputSet -- a.references).nonEmpty =>
       a.copy(child = prunedChild(child, a.references))
-    case w @ Window(_, _, _, _, child) if (child.outputSet -- w.references).nonEmpty =>
```

Comment:
First, `w.outputSet` always include `child.outputSet`.

Second, `w.references` only include the expressions present in the current `Window` operator. This set does not include attributes that are implicitly referenced by being passed through to the output tuple. Thus, it is not valid now. It will wrongly prune the child, if we keep it. Please correct me if my understanding is wrong.